libs/@blockprotocol/graph/src/stdlib/interval.ts

```diff
@@ -0,0 +1,333 @@
+ * if adjacent, the two intervals should span another given interval.
+ *
+ * @param lhs - The first interval of the comparison (order is unimportant)
+ * @param rhs - The second interval of the comparison
+ */
+export const intervalIsAdjacentToInterval = (
+  lhs: NonNullTimeInterval,
+  rhs: NonNullTimeInterval,
+): boolean => {
+  /*
+   Examples           |     1     |     2     |     3     |     4     |     5
+   ===================|===========|===========|===========|===========|===========
+   Interval A         | [---]     | [---)     | [---]     | [---)     | [-]
+   Interval B         |     (---] |     [---] |     [---] |     (---] |      [--]
+   -------------------|-----------|-----------|-----------|-----------|-----------
+   Contains Interval  |   true    |   true    |   false   |   false   |   false
+   */
+  return (
+    boundIsAdjacentTo(lhs.end, rhs.start) ||
+    boundIsAdjacentTo(lhs.start, rhs.end)
+  );
+};
+
+/**
+ * Returns whether or not the `rhs` {@link NonNullTimeInterval} is *completely contained* within the `lhs`
+ * {@link NonNullTimeInterval}.
+ *
+ * @param {NonNullTimeInterval} lhs - Checked if it contains the other
+ * @param {NonNullTimeInterval} rhs - Checked if it's contained _within_ the other
+ */
+export const intervalContainsInterval = (
+  lhs: NonNullTimeInterval,
+  rhs: NonNullTimeInterval,
+): boolean => {
+  /*
+   Examples           |     1     |     2     |     3     |     4     |     5     |     6
+   ===================|===========|===========|===========|===========|===========|===========
+   Interval A         |  [------] |   [----]  |   [----]  |   (----]  |   (----]  | [--]
+   Interval B         |    [--]   |   [---]   |   (---]   |   (---]   |   [---]   |   [---]
+   -------------------|-----------|-----------|-----------|-----------|-----------|-----------
+   Contains Interval  |   true    |   true    |   true    |   true    |   false   |   false
+   */
+  return (
+    compareBounds(lhs.start, rhs.start, "start", "start") <= 0 &&
+    compareBounds(lhs.end, rhs.end, "end", "end") >= 0
+  );
+};
+
+/**
+ * Returns whether or not the given {@link Timestamp} falls within the span of a given {@link NonNullTimeInterval}.
+ *
+ * @param {NonNullTimeInterval} interval
+ * @param {Timestamp} timestamp
+ */
+export const intervalContainsTimestamp = (
+  interval: NonNullTimeInterval,
+  timestamp: Timestamp,
+): boolean => {
+  /*
+   Examples            |     1     |     2     |     3     |     4
+   ====================|===========|===========|===========|===========
+   Interval            |    [----] |   (--]    |  [--)     | [--]
+   Timestamp           |      .    |      .    |     .     |       .
+   --------------------|-----------|-----------|-----------|-----------
+   Contains Timestamp  |   true    |   true    |   false   |   false
+   */
+  return (
+    compareBounds(
+      interval.start,
+      { kind: "inclusive", limit: timestamp },
```

Comment:
Could we move this out of the `return` statement? Then, we can reuse it and it's more readable.

Reply:
I have done so, although I don't think it really affected readability much 😅